### PR TITLE
[RUMF-1046] make Tracekit stoppable

### LINKED ISSUE #541: SDK override existing installed `onunhandledrejection` handler
Hi, 

I have a project where I've installed both Datadog RUM and Sentry (https://sentry.io) and when Sentry is initialized _before_ RUM, RUM will override the `onunhandledrejection` handler (see the [code](https://github.com/DataDog/browser-sdk/blob/8f86472688dbb3cf117ad422c11cd5ba7fa5aab8/packages/core/src/tracekit.ts#L312-L331)) 

The fix for this project was to initialize Datadog before Sentry, so Sentry calls both its handler & the previously installed handler (RUM). I'm wondering if the RUM team will be receptive of a PR changing the installation behavior to respect previously installed handlers. Thanks!

### PR DESCRIPTION
**Pull request by BenoitZugmeyer**
## Motivation

In the same spirit as #1133 and #1124 , this PR allows to stop/unpatch the tracekit instrumentation.

By using the `instrumentMethod` helper, it also fixes #541 (superseds #542)

## Changes

* Simplify tracekit tests a bit to make them synchronous and factorize the subscribe/unsubscribe in before/afterEach hooks
* Make Tracekit stoppable by using `instrumentMethod`
* Add a non-regression test for #541 collocated to the similar test for `onerror`

## Testing

Unit, manual, staging

---

I have gone over the [contributing](https://github.com/DataDog/browser-sdk/blob/main/CONTRIBUTING.md) documentation.

### CODE REVIEW

File: packages/core/src/domain/tracekit/tracekit.ts [hunk 1]
```diff
@@ -1,225 +1,104 @@
...
-    } catch (inner) {
-      exception = inner
-    }
-  })
-  if (exception) {
-    throw exception
+  return {
+    stop: () => {
+      stopInstrumentingOnError()
+      stopInstrumentingOnUnhandledRejection
```
Comment (bcaudan) [inline packages/core/src/domain/tracekit/tracekit.ts:48]:
```suggestion
      stopInstrumentingOnUnhandledRejection()
```

File: packages/core/src/domain/tracekit/tracekit.spec.ts [hunk 2]
```diff
@@ -1,165 +1,110 @@
...
-        expect(stack.name).toEqual('ReferenceError')
-        expect(stack.message).toEqual('foo is undefined')
-        unsubscribe(subscriptionHandler!)
-        done()
-      }
-      subscribe(subscriptionHandler)
-      traceKitWindowOnError('ReferenceError: foo is undefined', 'http://example.com', testLineNo)
+    it('should separate name, message for default error types (e.g. ReferenceError)', () => {
+      window.onerror!('ReferenceError: foo is undefined', 'http://example.com', testLineNo)
+      const [stack, ,] = callbackSpy.calls.mostRecent().args
```
Comment (bcaudan) [inline packages/core/src/domain/tracekit/tracekit.spec.ts:37]:
```suggestion
      const [stack] = callbackSpy.calls.mostRecent().args
```
or do we need it?
(other occurrences with or without)

File: packages/core/src/domain/tracekit/tracekit.ts (hunk 1 above)
Comment (BenoitZugmeyer) [reply packages/core/src/domain/tracekit/tracekit.ts:48]: How did I miss that!

File: packages/core/src/domain/tracekit/tracekit.spec.ts (hunk 2 above)
Comment (BenoitZugmeyer) [reply packages/core/src/domain/tracekit/tracekit.spec.ts:37]: Unneeded, I'll remove those.

**Review by bcaudan — APPROVED** [review]
Nice!

File: packages/core/src/domain/tracekit/tracekit.ts [hunk 3]
```diff
@@ -1,225 +1,104 @@
...
-      handlers.splice(i, 1)
-    }
-  }
-
-  if (handlers.length === 0) {
-    uninstallGlobalHandler()
-    uninstallGlobalUnhandledRejectionHandler()
-  }
-}
+export function startUnhandledErrorCollection(callback: UnhandledErrorCallback) {
```
Comment (amortemousque) [inline packages/core/src/domain/tracekit/tracekit.ts:41]: Like it!

**Review by amortemousque — APPROVED** [review]

**Pull request merged**